src/components/LibraryContainer.tsx

```diff
@@ -20,25 +23,53 @@ export interface LibraryContainerProps {
 }
 
 export interface LibraryContainerStates {
-    inSearchMode: boolean
+    inSearchMode: boolean,
+    searchText: string,
+    displayMode: displayMode,
+    selectedCategories: string[],
+    structured: boolean,
+    detailed: boolean
 }
 
 export class LibraryContainer extends React.Component<LibraryContainerProps, LibraryContainerStates> {
 
+    timeout: number;
     generatedSections: ItemData[] = null;
+    renderedSections: JSX.Element[] = null;
     searchCategories: string[] = [];
+    searcher: Searcher = null;
 
     constructor(props: LibraryContainerProps) {
         super(props);
-        this.state = { inSearchMode: false };
-        this.generatedSections = buildLibrarySectionsFromLayoutSpecs(this.props.loadedTypesJson, this.props.layoutSpecsJson, 
-        this.props.defaultSectionString, this.props.miscSectionString);
+        this.generatedSections = buildLibrarySectionsFromLayoutSpecs(this.props.loadedTypesJson, this.props.layoutSpecsJson,
+            this.props.defaultSectionString, this.props.miscSectionString);
 
         // Obtain the categories from each section to be added into the filtering options for search
         for (let section of this.generatedSections) {
             for (let childItem of section.childItems)
                 this.searchCategories.push(childItem.text);
         }
+
+        // Render the default view of the library. This property is displayed whenever the user is not doing a search
+        let index = 0;
+        this.renderedSections = this.generatedSections.map(data => <LibraryItem key={index++} libraryContainer={this} data={data} />);
+
+        // Initialize the search utilities
+        this.searcher = new Searcher({
+            onSearchModeChanged: this.onSearchModeChanged.bind(this),
+            libraryContainer: this,
+            sections: this.generatedSections,
+            categories: this.searchCategories
+        })
+
+        this.state = {
+            inSearchMode: false,
+            searchText: '',
+            displayMode: "list",
+            selectedCategories: this.searchCategories,
+            structured: false,
+            detailed: false
+        };
```

Comment:
There are [many changes for this file](https://github.com/DynamoDS/librarie.js/pull/61/files) also, I think the best way to do things would be closing this PR, and start the work again based on the latest code. Does that make sense?